WooCommerce/Classes/POS/Presentation/Card Present Payments/PointOfSaleCardPresentPaymentEventPresentationStyle.swift

```diff
@@ -69,7 +69,11 @@ extension CardPresentPaymentEventDetails {
                     endSearchAction: endSearch)))
 
         case .updateProgress(let requiredUpdate, let progress, let cancelUpdate):
-            return .alert(.updatingReader(viewModel: CardPresentPaymentUpdatingReaderAlertViewModel()))
+            return .alert(.updatingReader(
```

Comment:
If we do have multiple alerts/view models, we can decide which one to send here.

Reply:
Lemme know if you think the latest check could be further improved from https://github.com/woocommerce/woocommerce-ios/pull/13085/commits/9fb3cfbb3f8af24f225934f77db3556b4ec4aca6.

Reply:
Looks much better.

The only bit I don't love is the optional cancellation handlers that I commented on, and the `dismiss()` calls from the views when the handler's nil.

There's a risk that we dismiss the alert but don't cancel the update/connection, and leave the reader in a partially connected state, or the app out of sync with it. Perhaps we need to add a `cancelConnection` or `reset` function to the `CardPresentPaymentService` and call that? This seems like something to just log on a ticket and actually test, though.